x/superfluid/keeper/epoch.go

```diff
@@ -126,6 +129,52 @@ func (k Keeper) UpdateOsmoEquivalentMultipliers(ctx sdk.Context, asset types.Sup
 
 		multiplier := k.calculateOsmoBackingPerShare(pool, osmoPoolAsset)
 		k.SetOsmoEquivalentMultiplier(ctx, newEpochNumber, asset.Denom, multiplier)
+	} else if asset.AssetType == types.SuperfluidAssetTypeConcentratedShare {
+		// LP_token_Osmo_equivalent = OSMO_amount_on_pool / LP_token_supply
+		poolId := cltypes.MustGetPoolIdFromShareDenom(asset.Denom)
+		pool, err := k.clk.GetPoolFromPoolIdAndConvertToConcentrated(ctx, poolId)
+		if err != nil {
+			k.Logger(ctx).Error(err.Error())
+			k.BeginUnwindSuperfluidAsset(ctx, 0, asset)
```

Comment:
Calling these methods should be explicitly tested. This seems to be a piece of logic that could be fatal

Reply:
Added test here https://github.com/osmosis-labs/osmosis/pull/4884/commits/88074a0998d28a231c4306cc5b3d831b139eda4d